src/Microsoft.Dnx.Watcher.Core/Impl/ProjectProvider.cs

```diff
@@ -12,19 +12,28 @@ public class ProjectProvider : IProjectProvider
     {
         public bool TryReadProject(string projectFile, out IProject project, out string errors)
         {
+            errors = null;
+            project = null;
+
             Runtime.Project runtimeProject;
             if (!TryGetProject(projectFile, out runtimeProject, out errors))
             {
-                project = null;
                 return false;
             }
 
-            errors = null;
-            project = new Project(runtimeProject);
+            try
+            {
+                project = new Project(runtimeProject);
+            }
+            catch (Exception ex)
+            {
+                errors = ex.Message;
```

Comment:
You should also include messages from the inner exception (if any). Also, you might want to handle `AggregateExceptions` as a special case